test/core.jl

```diff
@@ -690,6 +690,16 @@ let A = [1]
     @test x == 1
 end
 
+# Pointer finalizer (issue #15408)
+let A = [1]
+    testlib = Libdl.dlopen("libccalltest")
+    ccall(Libdl.dlsym(testlib, "set_c_int"), Void, (Cint,), Cint(1))
```

Comment:
`ccall((:set_c_int, "libccalltest"))` and `cglobal((:finalizer_cptr, "libccalltest"))` See the [ccall test](https://github.com/JuliaLang/julia/blob/4b85d85a9f7256718bbba7a45fc68fb454dd0532/test/ccall.jl). This test could probably be moved there too.